src/app/staking/[addressOrENSName]/page.jsx

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import Tenant from "@/lib/tenant/tenant";
+import { apiFetchStakedDeposits } from "@/app/api/staking/getStakedDeposits";
+import { fetchDelegate } from "@/app/api/common/delegates/getDelegates";
+import { HStack } from "@/components/Layout/Stack";
+import { DepositList } from "@/app/staking/[addressOrENSName]/deposits/DepositList";
+import { StakingStats } from "@/app/staking/components/StakingStats";
+import StakingFaq from "@/app/staking/components/StakingFaq";
+import { PanelClaimRewards } from "@/app/staking/components/PanelClaimRewards";
+import { PanelNewDeposit } from "@/app/staking/components/PanelNewDeposit";
+import { resolveENSName } from "@/app/lib/ENSUtils";
+import { revalidatePath } from "next/cache";
+import { StakingIntro } from "@/app/staking/components/StakingIntro";
+import { PanelSetAllowance } from "@/app/staking/components/PanelSetAllowance";
+
+async function fetchDeposits(address) {
+  "use server";
+  return apiFetchStakedDeposits({ address });
+}
+
+async function apiFetchDelegate(address) {
+  "use server";
+  return fetchDelegate(address);
+}
+
+export default async function Page({ params: { addressOrENSName } }) {
+  const address = (await resolveENSName(addressOrENSName)) || addressOrENSName;
+
+  const { token } = Tenant.current();
+  const { ui, contracts } = Tenant.current();
+  const deposits = await fetchDeposits(address.toLowerCase());
+
+  if (!ui.toggle("staking")) {
+    return <div>Route not supported for namespace</div>;
+  }
+
+  const [totalSupply, totalStaked, rewardPerToken, rewardDuration] =
+    await Promise.all([
+      contracts.token.contract.totalSupply(),
+      contracts.staker.contract.totalStaked(),
+      contracts.staker.contract.rewardPerTokenAccumulated(),
+      contracts.staker.contract.REWARD_DURATION(),
+    ]);
+
+  const hasDeposits = deposits && deposits.length > 0;
+
+  return (
+    <HStack className="grid grid-cols-1 grid-rows-2 sm:grid-cols-4 sm:grid-rows-1 gap-5 sm:gap-10 mt-12">
+      <div className="sm:col-span-4">
+        {hasDeposits ? (
+          <div>
+            <div className="font-black text-2xl mb-5">
+              Your {token.symbol} Stake
+            </div>
+
+            <DepositList
+              deposits={deposits}
+              fetchDelegate={apiFetchDelegate}
+              refreshPath={async (path) => {
+                "use server";
+                revalidatePath(path);
+              }}
+            />
+          </div>
+        ) : (
+          <StakingIntro />
+        )}
+
+        <div className="mt-10">
+          <div className="font-black text-2xl mb-5">
+            {token.symbol} Staking Metrics
+          </div>
+          <StakingStats
```

Comment:
This is the stats component that's also in the airdrop, right? 
I really think it would be cool if we fleshed out that component library a bit